packages/services/src/epcisEvents/aggregationEvent.ts

```diff
@@ -0,0 +1,64 @@
+import { VerifiableCredential } from '@vckit/core-types';
+import { issueVC } from '../vckit.service.js';
+import { uploadJson } from '../storage.service.js';
+import { IdentificationKeyType, registerLinkResolver } from '../linkResolver.service.js';
+import { IService } from '../types/IService.js';
+import { IAggregationEvent, IAggregationEventContext } from './types';
+import { generateUUID } from '../utils/helpers.js';
+import { getIdentifierByObjectKeyPaths } from './helpers.js';
+import { validateAggregationEventContext } from './validateContext.js';
+import { EPCISBusinessStepCode, EPCISEventAction, EPCISEventDisposition, EPCISEventType } from '../types/epcis.js';
+
+export const processAggregationEvent: IService = async (aggregationEvent: IAggregationEvent, context: IAggregationEventContext): Promise<VerifiableCredential> => {
+  const validationResult = validateAggregationEventContext(context);
+  if (!validationResult.ok) {
+    throw new Error(validationResult.value);
+  }
+
+  const { vckit, epcisAggregationEvent, dlr, storage, identifierKeyPaths } = context;
+  const identifier: string = getIdentifierByObjectKeyPaths(aggregationEvent.data, identifierKeyPaths);
+  if (!identifier) {
+    throw new Error('Identifier not found');
+  }
+
+  const credentialSubject = {
```

Comment:
Should we build a common function for reusable in other events? Because I see in `epcis.service.ts` we also have one function return similar to this object?

Reply:
What is the function you mentioned in the `epcis.service.ts` file?

Reply:
You can check epcisTransformationCrendentialSubject() and see it return...